src/store/GlobalState.js

```diff
@@ -0,0 +1,155 @@
+import { createContext, useEffect, useReducer } from 'react';
+import { collection, doc, addDoc, deleteDoc, updateDoc, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+import appReducer from './appReducer';
+import { getCurrentMonth, getUKFormattedDate } from '../utils/helpers';
+
+const currentMonth = getCurrentMonth();
+
+const initialState = {
+  expenses: [],
+  selectedMonthExpenses: [],
+  currentMonth: currentMonth,
+  selectedMonth: currentMonth,
+  lastAddedExpense: {},
+  expenseToDelete: {},
+  expenseToEdit: {},
+  isConfirmMonthModalOpen: false,
+  isConfirmDeleteModalOpen: false,
+  isEditExpenseModalOpen: false,
+  didErrorOccur: false,
+  isAlertOpen: false,
+  userAction: '',
+};
+
+export const GlobalContext = createContext(initialState);
```

Comment:
is it really necessary to pass the initial state here?

it really confused me quite a lot.